src/test/ui/closure-struct-diagnostics.rs

```diff
@@ -0,0 +1,20 @@
+// Check that closures do not implement `Copy` if their environment is not `Copy`.
+
```

Comment:
Can you move this into `ui/closures/2229_closure_analysis/closure_origins` and add test cases for 
1. Tuple being captured (different from Tuple Struct)
2. SingleVariant Enum
3. MultiVariant Enum

and  change the reason for the tests to ensure that precise paths are reported back for a when reporting closure origins. 

Also mix the tests to be this and the expect_fn one.